### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by KerronKing**
In this project we:

- Added Webpack configuration files and directory skeleton.
- Created Ship and Player factories.
- Create DOM render functions for the gameboard.
- Wrote functions for creating players and computer ships.
- Wrote the game loop.
- Added jest tests along the way.
- Edited game styling.

### CODE REVIEW

File: src/ship.js
Comment (shloch) [inline src/ship.js]: - this class should be able to tell if ship has been sunk [[ see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/battleship/sunk_requirement.png) ]

- this should have a `hit()` method that takes a number as an argument and marks that position as hit [[see requiredments](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/battleship/hit_requirement.png)]

File: src/gameboard.js
Comment (shloch) [inline src/gameboard.js]: - Gameboards should have a receiveAttack function that takes a pair of coordinates .... [ [see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/battleship/receiveattack_requirement.png) ]
This method should eventually call the `hit()` function described above

- this file should also have a way of keeping tracks of missed attacks
[ [see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/battleship/missed_attacks_requirement.png) ]

**Review by shloch — CHANGES_REQUESTED** [review]
Hello @KerronKing 
You game works perfectly....but you seem to have not followed the project requiredments prescribed by microverse, found in [Odin project](https://www.theodinproject.com/courses/javascript/lessons/battleship). 

It will be good for organisational purpose to follow through with the steps here. They are organised in such a manner that we grasp js some concepts better, a coherent industry-standard thought-process; review comes easy too. It equally makes us ensure that you followed through the instructions. 

Your project works, but it seems like you followed another guideline, you have a different organisation apart of file names. Please follow the instructions. I just mentioned some observations below :

### { waiting for another review with re-organisation }

NB : MOST OF YOUR METHODS ARE VERY USEFUL SO YOU'LL JUST BE CONCERNED AROUND RE-ORGANISING YOUR CODE AROUND THE PROJECT REQUIREMENTS 

HAPPY CODING 😊😊😊😊😊

**Comment by shloch** [comment]
Hello once small @KerronKing , after undergoing a thorough review with your code.....it's great thing that it mights the functionality goals. It's such a good thing for CREATIVITY. However this project was meant to follow guidelines.

We've reserved capstone projects for creativity. So while we are for the work you've produced so far, we plead with you to approach this particular one following guidelines provided

Thanks

File: src/ship.js
Comment (KerronKing) [reply src/ship.js]: isSunk() and hit() functions were successfully implemented.

File: src/gameboard.js
Comment (KerronKing) [reply src/gameboard.js]: Gameboard was converted to a factory, receiveAttack() was appropriately implemented, taking 2 coordinate inputs and Gameboard tracks missed attacks.

**Review by bolah2009 — APPROVED** [review]
Awesome design and game flow with [the app](https://rawcdn.githack.com/KerronKing/Battleship-JS/56aac3cd696ecbe534c9d4430eeab7607ba826b0/dist/index.html)

It's a 👍🏾 from me!

### :heavy_check_mark: Approved!

Nice job implementing the suggested changes.

Good luck with your next projects. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Pull request merged**
